src/coverlet.collector/InProcDataCollection/CoverletInProcDataCollector.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Reflection;
+using Coverlet.Core.Instrumentation;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollector.InProcDataCollector;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.InProcDataCollector;
+
+namespace Coverlet.Collector.InProcDataCollector
+{
+    public class CoverletInProcDataCollector : InProcDataCollection
+    {
+        public void Initialize(IDataCollectionSink dataCollectionSink)
+        {
+        }
+
+        public void TestCaseEnd(TestCaseEndArgs testCaseEndArgs)
+        {
+        }
+
+        public void TestCaseStart(TestCaseStartArgs testCaseStartArgs)
+        {
+        }
+
+        public void TestSessionEnd(TestSessionEndArgs testSessionEndArgs)
+        {
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                Type injectedInstrumentationClass = GetInstrumentationClass(assembly);
+                if (injectedInstrumentationClass is null)
+                    continue;
+
+                var unloadModule = injectedInstrumentationClass.GetMethod(nameof(ModuleTrackerTemplate.UnloadModule), new[] { typeof(object), typeof(EventArgs) });
+                if (unloadModule is null)
+                    continue;
+
+                try
+                {
+                    unloadModule.Invoke(null, new[] { null, EventArgs.Empty });
+                }
+                catch
+                {
+                    // Ignore exceptions and continue with the unload
```

Comment:
Should we log something?Is this correct?I mean we shouldn't have exception on saving hits to file cc: @tonerdo

Reply:
Added logs.